streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java

```diff
@@ -246,4 +246,11 @@ default boolean commitRequested() {
      * @return This returns the time the task started idling. If it is not idling it returns empty.
      */
     Optional<Long> timeCurrentIdlingStarted();
+
+    /**
+     * Gets the exception that caused the failure of the task.
+     *
+     * @return exception that caused the failure of the task
+     */
+    Optional<RuntimeException> getException();
```

Comment:
I think we can narrow down the scope of this exception, if it is only going to be used during restoration time. But nevertheless we can discuss about this later.